File: reuse_java/src/test/java/com/kmmania/DistanceRunnerTest.java
package com.kmmania;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.*;
import static org.mockito.Mockito.*;

public class DistanceRunnerTest {

    @Test
    @DisplayName("display() method should display correct information")
    void displayShouldDisplaysCorrectInformation() {
        // Given
        String firstName = "Eliud";
        String lastName = "Kipchoge";
        String speciality = "marathon";
        AthleteInterface athleteMock = mock(AthleteInterface.class);
        when(athleteMock.mySpeciality()).thenReturn("My speciality is " + speciality + ".");
        DistanceRunner distanceRunner = new DistanceRunner(firstName, lastName, speciality);
        distanceRunner.setAthlete(athleteMock);

        // When
        distanceRunner.display();

        // Then
        assertThat(distanceRunner.getFirstName()).isEqualTo(firstName);
        assertThat(distanceRunner.getLastName()).isEqualTo(lastName);
        // verify(athleteMock).mySpeciality();
    }
}

File: reuse_kotlin/src/test/kotlin/DistanceRunnerTest.kt
import com.kmmania.AthleteInterface
import com.kmmania.DistanceRunner
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*

class DistanceRunnerTest {

    @Test
    fun `display() method should display correct information`() {
        // Given
        val firstName = "Eliud"
        val lastName = "Kipchoge"
        val speciality = "marathon"
        val athleteMock = mock(AthleteInterface::class.java)
        `when`(athleteMock.mySpeciality()).thenReturn("My speciality is $speciality.")
        val distanceRunner = DistanceRunner("Eliud", "Kipchoge", athleteMock)

        // When
        distanceRunner.display()

        // Then
        assertThat(distanceRunner.firstName).isEqualTo(firstName)
        assertThat(distanceRunner.lastName).isEqualTo(lastName)
        verify(athleteMock).mySpeciality()
    }

}